src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1512,3 +1512,42 @@ def list_consumption_locations():
     client = web_client_factory()
     regions = client.list_geo_regions(sku='Dynamic')
     return [{'name': x.name.lower().replace(" ", "")} for x in regions]
+
+
+def enable_zip_deploy(resource_group_name, name, src, slot=None):
+    # if src is not of .zip extension throw exception
+    if not src.endswith('.zip'):
+        raise CLIError("src '{}' path needs to be .zip file".format(src))
+
+    client = web_client_factory()
+    user_name, password = _get_site_credential(client, resource_group_name, name)
+    scm_url = _get_scm_url(resource_group_name, name, slot)
+    zip_url = scm_url + '/api/zipdeploy'
+
+    import urllib3
+    try:
+        import urllib3.contrib.pyopenssl
+        urllib3.contrib.pyopenssl.inject_into_urllib3()
+    except ImportError:
+        pass
```

Comment:
Could you shed some light on this code snippet? What does it try to accomplish and in what situation the import will fail?

Reply:
Seems to do with http://urllib3.readthedocs.io/en/latest/reference/urllib3.contrib.html#module-urllib3.contrib.pyopenssl but would still like clarification.

Reply:
Honestly i added this since only other place where this library was being imported and used was checking and throwing this exception. I believe Yugang added it so he can probably shed light on it.

---
In reply to: [145754736](https://github.com/Azure/azure-cli/pull/4700#discussion_r145754736) [](ancestors = 145754736)

Reply:
Ok, I think at least we can put the URL to the comments. The article does explain the cause.

@derekbekoe @yugangw-msft why isn't this enabled globally. Aren't other places need SSL certs verification?

Reply:
We can remove this. This workaround was only used by trace streaming to prevent the last line loss. 
Also, for the next line you can directly use dictionary to construct headers rather using urllib3, so that we can remove the `import urllibs`